__tests__/app.test.js

```diff
@@ -183,3 +183,83 @@ describe("CORE: GET - /api/articles/:article_id", () => {
       });
     })
  
+
+
+
+   
+    
+
+describe("CORE: POST /api/articles/:article_id/comments", () => {
+  test("201 : Endpoint displays the inserted comment with correct properties/values", () => {
+    const input = {username : 'rogersop', body : 'big comment' }
+    return request(app)
+      .post("/api/articles/4/comments")
+      .send(input)
+      .expect(201)
+      .then(({body}) => {
+          expect(body).toHaveProperty("comment_id", expect.any(Number));
+          expect(body).toHaveProperty("votes", (0)); 
+          expect(body).toHaveProperty("created_at", expect.any(String)); 
+          expect(body).toHaveProperty("author", ('rogersop')); 
+          expect(body).toHaveProperty("body", ('big comment')); 
+          expect(body).toHaveProperty("article_id", (4)); 
+        });
+      })
+      test('Receive 400 error - User doesnt exist on users table', () => {
+        const badInput = {username : 'cheeseman', body : 'big comment' }
+        return request(app)
+            .post("/api/articles/4/comments")
+            .send(badInput)
+            .expect(400)
+            .then(({body}) => {
+                expect(body).toEqual({Error: 'User does not Exist'})
+            })
+            
+  })
+  test('Receive 400 error - Body requires not null value', () => {
+    const badInput2 = {username : 'rogersop', body : null }
+    return request(app)
+        .post("/api/articles/4/comments")
+        .send(badInput2)
+        .expect(400)
+        .then(({body}) => {
+            expect(body).toEqual({Error : 'Comment requires a text input'})
+        })
+})
+test('400: ID outside of data range outputs a useful error message', () => {
+  const input = {username : 'rogersop', body : 'big comment' }
+  return request(app)
+  .post("/api/articles/8365298364982642/comments")
+  .send(input)
+  .expect(400)
+  .then(({body}) => {
+    expect(body).toEqual({Error: "400, Invalid ID"})
+  })
+})
+test('400  -  Incorrect url parameter input outputs a useful error message', () => {
```

Comment:
Again with the description we could be more explicit, the problem here is that the id is invalid, not that the url is invalid.